runtime/drivers/duckdb/connectors.go

```diff
@@ -77,30 +81,38 @@ func (c *connection) ingestLocalFiles(ctx context.Context, env *connectors.Env,
 	// Not using query args since not quite sure about behaviour of injecting table names that way.
 	// Also, it's a source, so the caller can be trusted.
 
-	var from string
-	if conf.Format == ".csv" && conf.CSVDelimiter != "" {
-		from = fmt.Sprintf("read_csv_auto(['%s'], delim='%s')", path, conf.CSVDelimiter)
-	} else {
-		from, err = getSourceReader(localPaths)
-		if err != nil {
-			return err
-		}
+	from, err := getSourceReader(localPaths, conf)
+	if err != nil {
+		return err
 	}
 
 	qry := fmt.Sprintf("CREATE OR REPLACE TABLE %s AS (SELECT * FROM %s)", source.Name, from)
 
 	return c.Exec(ctx, &drivers.Statement{Query: qry, Priority: 1})
 }
 
-func getSourceReader(paths []string) (string, error) {
-	ext := fileutil.FullExt(paths[0])
-	if ext == "" {
+func getSourceReader(paths []string, config *connectors.Config) (string, error) {
```

Comment:
Let's simplify this to just `getSourceReader(paths []string, csvDelimiter string) (string, error)` to reduce dependencies.

If in future we need to add more CSV options, we can change it to a `csvOpts` struct param.

Reply:
You're right. The format option is supposed to be a way for the user to force a specific format (in case file extension is weird), so it's optional (hence the check for path extension).